### ISSUE TITLE
[Feature request] A more compact frequency display in merged mode

### ISSUE THREAD

**Issue by Pintliz** [FooSoft/yomichan#1093]
![image](https://user-images.githubusercontent.com/24824399/101930405-1340ef80-3c13-11eb-9cbc-10ada0c57e86.png)
Currently both dictionary names and entry names are repeatedly displayed. Would be nice if one of them can be condensed. Considering dictionary names are usually longer than entry names they are probably the ideal target.

something like:
[(dictionary name):entry1:frequency1 | entry2:frequency2] [(dictionary name2):entry1:frequency1 | entry2:frequency2]

**Comment by Thermospore** [comment]
What is the difference between the 文 and the 綾 entries?

**Comment by Pintliz** [comment]
most of the time it's written as あや,アヤ or 綾. 
文(あや) is kinda rare.
the frequency displayed here probably included ぶん and もん

**Comment by Thermospore** [comment]
Ohhhh, gotcha. I use a yomichan in a different mode; it looked like that was part of the dictionary name

**Comment by Thermospore** [comment]
In any case, I'm probably going to go in and manually abbreviate my freq list names (eg `IC`, `A&J`) since I have a lot of them

**Comment by Thermospore** [comment]
Actually on topic:

Maybe behavior should change depending on if the frequency list distinguishes between readings or not. For example:
![image](https://user-images.githubusercontent.com/35350819/102687637-ed7c9300-419c-11eb-8182-5b3224062572.png)
The A&J, Innocent Corpus, and Wikipedia freq lists do not distinguish between readings, and only look at the form. Thus:
 - The reading does not need to be displayed in the frequency tag
 - There only needs to be one tag per form

So for example, A&J would only have 2 entries. one for 洞穴, ほら穴, and neither would include the reading. You could even collapse both of those into a single tag so you don't have to display the name "Anime & Jdrama" twice, like @Pintliz suggests.

BCCWJ would be left as is, since it distinguishes between readings (just collapse, like @Pintliz suggests)

Extreme example
![image](https://user-images.githubusercontent.com/35350819/102687591-9676be00-419c-11eb-8fa0-38b530466d9a.png)
this could be reduced down to `3 forms`*`the 3 readingless-dictionaries` + `the one reading-conscious entry from BCCWJ`, then collapsed to show each dictionary name once

**Comment by Thermospore** [comment]
Here is another thing that can be collapsed (this applies to all term grouping modes):
![image](https://user-images.githubusercontent.com/35350819/102688990-523ceb00-41a7-11eb-9e3d-9866d12e6017.png)
You could reduce this to something like `BCCWJ(short):451, 20834, 100044` (there are different entries for different parts of speech, I believe)

**Comment by toasted-nutbread** [comment]
Honestly I think this is running into the same issue that I initially ran into when deciding how to present pitch accents. I had initially planned to display them as tags, but it was just too cluttered and made the information hard to understand. The same issue seems to be present here, particularly when using multiple frequency dictionaries; I noticed that @Thermospore even decided to recompile the dictionaries [with shortened names](https://github.com/FooSoft/yomichan/issues/855#issuecomment-750201517), presumably due to this issue.

So I'm wondering if we should rethink how frequency tags are presented altogether, maybe something more similar to pitch accents. This would allow for significantly reducing the amount of repeated data shown, specifically the repeated dictionary names. I'm worried that packing too much information into a tag will have the effect of making it no longer function as a "tag" (short, brief information), and refactoring the design would allow better clarification of what the numbers + disambiguation text means.

**Comment by Thermospore** [comment]
> I noticed that @Thermospore even decided to recompile the dictionaries [with shortened names](https://github.com/FooSoft/yomichan/issues/855#issuecomment-750201517), presumably due to this issue.

Correct! Another idea to help with readability for the meantime is perhaps some custom CSS could be used to manually set a color for each freq dictionary

> So I'm wondering if we should rethink how frequency tags are presented altogether, maybe something more similar to pitch accents. This would allow for significantly reducing the amount of repeated data shown, specifically the repeated dictionary names.

This sounds like a great idea. I don't have many concrete ideas to offer (at the moment), but I'm happy to provide feedback and I'm interested to see where this goes!

**Comment by toasted-nutbread** [comment]
Here are a few examples of a new format that is more similar to how pitch accents are presented. There are some potential improvements with word wrapping, but the idea is the same. Some small headings are added to differentiate the different sections, which I am not sure are necessary or not, but the different list/numbering schemes could be confusing otherwise.

## Current format
![current](https://user-images.githubusercontent.com/11037431/103962138-34b6cd80-5124-11eb-9d76-db19b3692950.png)

## Current format (compact layout)
![current-compact](https://user-images.githubusercontent.com/11037431/103962146-3b454500-5124-11eb-8978-ce5985f2139c.png)

## New format
![new](https://user-images.githubusercontent.com/11037431/103962169-4d26e800-5124-11eb-945c-e8a60c4d72d3.png)

## New format (compact layout)
![new-compact](https://user-images.githubusercontent.com/11037431/103962181-544df600-5124-11eb-8176-1dc83a830324.png)

**Comment by Thermospore** [comment]
Wow, nice!

The new format is clearly preferable for entries with multiple forms/readings, where the original format turns into tag spaghetti (such as in your examples above, or the dreaded 虚ウロ)

There is one situation where I think the original format comes out on top though, and that is for entries with only one form/reading:
![image](https://user-images.githubusercontent.com/35350819/103968148-c7993d00-510f-11eb-9688-ebff85842d15.png)
The frequencies can all fit neatly and readably in one line with the original format, where as the new format would take up one line per freq dictionary

I'd have to pay attention, but I'd guess about 50% of the entries I look up only have one form/reading

Maybe for simple entries it could compact down to one line?

Apart from that, it looks like massive improvement and way more readable!

(to be fair, I'm gonna remove `Bs` and `B長` once I'm done looking out for irregularities, so I'll "only" have 4 haha)

**Comment by toasted-nutbread** [comment]
I suppose there's no perfect way to have the list be maximally compact and have the meaning be clear; in your last example, the titles are abbreviated from their longer form, which would increase the line count for a standard user. Word wrapping is the major issue with trying to compress it further, something like the following makes the tags a bit unclear, and it's often worse in more complex cases.
![image](https://user-images.githubusercontent.com/11037431/103972130-54f18700-513a-11eb-9664-7f5a81e2afb9.png)
![image](https://user-images.githubusercontent.com/11037431/103972202-85392580-513a-11eb-9c83-8f04884dcc92.png)
![image](https://user-images.githubusercontent.com/11037431/103972258-9da94000-513a-11eb-8647-788755e3eb00.png)

**Comment by Thermospore** [comment]
For ease of discussion, I'll define
 - case A: entries with multiple forms and or multiple readings (like in all of your screenshots)
 - case B: entries with a single form and a single reading (like in the `国語` screenshot. or whenever not using `group related terms` mode)

> something like the following makes the tags a bit unclear, and it's often worse in more complex cases.

I agree, I think case A works better on multiple lines, as in the initial `New format` screenshots!

I only think the frequency entries should be collapsed to one line in case B. In this case no additional specification needs to be given (such as `(黄いろい only)`, `(一所:いっしょ only)`, etc) and all that needs to be displayed is `[freq list name]:[freq]`. This can be done cleanly as shown in the `国語` screenshot

Using `New format` with case B would end up like this, which has unnecessary deadspace and newlines:
![image](https://user-images.githubusercontent.com/35350819/103986251-e3fca000-5136-11eb-90bb-305bcdd47e9c.png)

> in your last example, the titles are abbreviated from their longer form, which would increase the line count for a standard user.

Before I abbreviated my names, I always fit 3 freq entries per line at this window size:
![paste-e7e0c662b4336a68507c6fa88cc539561e202c9d](https://user-images.githubusercontent.com/35350819/103986753-c3811580-5137-11eb-9fc7-9f04e1b2c5d2.jpg)
The standard user probably wouldn't have more than 3, but even then they would still benefit. By keeping the `Current format` for case B, even 6 unabbreviated freq entries are reduced from 6 lines down to 2 while maintaining great readability (again, I don't suggest this format for case A!).

> Word wrapping is the major issue with trying to compress it further

Hmmm good point

For case B, I think word wrap was solved elegantly in `Current format` by simply keeping each `[freq list name]:[freq]` pair as an indivisible chunk

For case A, I think the `New format` is already pretty good, I don't have any suggestions tbh

In summary I think
 - case A: use the awesome `New format`
 - case B: keep the `Original format` (or at least its compact and chunked nature)

**Comment by toasted-nutbread** [comment]
New PR up which compresses entries onto a single line when possible and in compact mode. Should be included in the next release.
![image](https://user-images.githubusercontent.com/11037431/104083179-9b161b80-520a-11eb-8772-a97aebc975f3.png)

**Comment by Thermospore** [comment]
Sweet, looking forward to trying it out
